libtrap/src/ifc_file.c

```diff
@@ -637,14 +665,19 @@ int create_file_send_ifc(trap_ctx_priv_t *ctx, const char *params, trap_output_i
    free(dest);
    priv->filename_base_length = strlen(exp_result.we_wordv[0]);
 
-   priv->filename = (char *) calloc(priv->filename_base_length + 1, sizeof(char));
+   priv->filename = (char *) calloc(priv->filename_base_length + 7, sizeof(char));
    if (!priv->filename) {
       free(priv);
       wordfree(&exp_result);
       return trap_error(ctx, TRAP_E_MEMORY);
    }
 
-   strncpy(priv->filename, exp_result.we_wordv[0], priv->filename_base_length + 1);
+   time(&timer);
+   tm_info = localtime(&timer);
+   strftime(priv->filename, priv->filename_base_length + 7, exp_result.we_wordv[0], tm_info);
+   priv->filename_base_length = strlen(priv->filename);
+   _mkdir(priv->filename);
```

Comment:
Change the definition of _mkdir function to: "static int _mkdir(char *dir)" and after that, check the return value here and adjust the code behavior for all possible outcomes.